### PR TITLE
feat(extra): improve rulesToQuery

### PR DESCRIPTION
**Pull request by vonagam**
Now such ability will return a query object instead of null:
```js
cannot('read', 'Post')
can('read', 'Post')
```

Also simplified code.

### CODE REVIEW

**Comment by stalniy** [comment]
Thanks for the PR. I will need some time to review and release.

**Comment by stalniy** [comment]
By the way, this is the correct behavior:

```js
cannot('read', 'Post')
can('read', 'Post')
```

because rules order matters, you can read about this in [the docs](https://stalniy.github.io/casl/abilities/2017/07/20/define-abilities.html#combining-abilities)

**Comment by stalniy** [comment]
close as invalid, feel free to ask questions if something is unclear

**Comment by vonagam** [comment]
Ok, then you agree that the snippet should correspond to an empty query (as i do) and that rules order matters. The problem is that current implementation will return not an empty query, but a null ([as soon as it encounters](https://github.com/stalniy/casl/blob/491099b9a808c8253fb924de039b8367c3addfe5/packages/casl-ability/src/extra.js#L14) inverted rule without conditions). While my pull request fixes it and it will return an empty query.

Please look at the test in this pull request, it will fail with current code.

In summary, currently `cannot + can = cannot`, while it should be `cannot + can = can`.

**Comment by stalniy** [comment]
Make sense. I think that you haven’t seen my review comment until now (I submitted review)

File: packages/casl-ability/src/extra.js [hunk 1]
```diff
@@ -11,23 +10,18 @@ export function rulesToQuery(ability, action, subject, convert) {
 
     if (!rule.conditions) {
       if (rule.inverted) {
-        return null;
-      }
-
-      if (query[op]) {
+        break;
```
[the remark this answers is not in the document]
Comment (vonagam) [reply packages/casl-ability/src/extra.js]: We iterate in reverse order. 
So first one will be an inverted rule without conditions. 
We break. 
As there will be no `$or` queries the result will be `null`.
Also all current tests are passing and this case is present in tests.

**Comment by vonagam** [comment]
Added second test showing another problem that this PR fixes:

```js
cannot('read', 'Post', { private: true })
can('read', 'Post')
```

This should produce an empty query.

File: packages/casl-ability/src/extra.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (stalniy) [reply packages/casl-ability/src/extra.js]: Ah, I see. It was my mistake. I forgot that rules are prepended in the array inside `Ability` and then returned in reverse order. 

Thanks for highlighting this!

**Comment by stalniy** [comment]
published in `@casl/ability@3.1.1`

Again, thanks for the awesome work!
